pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java

```diff
@@ -71,37 +73,33 @@ public String getExpression() {
   }
 
   public AggregationInfo buildAggregationInfo() {
-    String identifier = null;
-    for (AstNode astNode : getChildren()) {
+    String expression;
+    // COUNT aggregation function always works on '*'
+    if (_name.equalsIgnoreCase("count")) {
+      expression = "*";
+    } else {
+      List<? extends AstNode> children = getChildren();
+      if (children == null || children.size() != 1) {
+        throw new Pql2CompilationException("Aggregation function expects exact 1 argument");
```

Comment:
We should think about removing this limitation. For example percentile(column, 95).

Reply:
We should enhance our compiler so that it can identify UDF vs Aggregation, and directly compile the query into final format.